assets/dev/js/editor/atomic-widgets/views/a-container.js

```diff
@@ -0,0 +1,347 @@
+		}
+	},
+
+	onRender() {
+		BaseElementView.prototype.onRender.apply( this, arguments );
+
+		// Defer to wait for everything to render.
+		setTimeout( () => {
+			this.droppableInitialize();
+		} );
+	},
+
+	droppableInitialize() {
+		this.$el.html5Droppable( this.getDroppableOptions() );
+	},
+
+	isDroppingAllowed() {
+		return true;
+	},
+
+	filterSettings( newItem ) {
+		if ( 'container' !== newItem?.elType ) {
+			return;
+		}
+
+		const parentContainer = this;
+
+		if ( parentContainer.isBoxedWidth() ) {
+			newItem.settings.content_width = 'full';
+		} else if ( 0 !== parentContainer.getNestingLevel() ) {
+			newItem.settings.content_width = 'full';
+		}
+	},
+
+	childViewOptions() {
+		return {
+			emptyViewOwner: this,
+		};
+	},
+
+	getCurrentUiStates() {
+		const currentDeviceMode = elementor.channels.deviceMode.request( 'currentMode' ),
+			deviceSuffix = 'desktop' === currentDeviceMode ? '' : '_' + currentDeviceMode,
+			directionSettingKey = this.getDirectionSettingKey() + deviceSuffix,
+			currentDirection = this.container.settings.get( directionSettingKey );
+
+		return {
+			directionMode: currentDirection || ContainerHelper.DIRECTION_DEFAULT,
+		};
+	},
+
+	onDeviceModeChange() {
+		SetDirectionMode.set( this.getContainer() );
+	},
+
+	getDirectionSettingKey() {
+		const containerType = this.container.settings.get( 'container_type' ),
+			directionSettingKey = 'grid' === containerType
+				? 'grid_auto_flow'
+				: 'flex_direction';
+
+		return directionSettingKey;
+	},
+
+	behaviors() {
+		const behaviors = BaseElementView.prototype.behaviors.apply( this, arguments );
+
+		_.extend( behaviors, {
+			// TODO: Remove. It's a temporary solution for the Navigator sortable.
+			Sortable: {
+				behaviorClass: require( 'elementor-behaviors/sortable' ),
+				elChildType: 'widget',
+			},
+			Resizable: {
+				behaviorClass: WidgetResizable,
+			},
+		} );
+
+		return elementor.hooks.applyFilters( 'elements/container/behaviors', behaviors, this );
+	},
+
+	onDestroy() {
+		BaseElementView.prototype.onDestroy.apply( this, arguments );
+
+		elementor.stopListening( elementor.channels.deviceMode, 'change', this.onDeviceModeChange );
+	},
+
+	/**
+	 * TODO: Remove. It's a temporary solution for the Navigator sortable.
+	 *
+	 * @return {{}} options
+	 */
+	getSortableOptions() {
+		// TODO: Temporary hack.
+		return {
+			preventInit: true,
+		};
+	},
+
+	getDroppableAxis() {
+		const isColumnDefault = ( ContainerHelper.DIRECTION_DEFAULT === ContainerHelper.DIRECTION_COLUMN ),
+			currentDirection = this.getContainer().settings.get( this.getDirectionSettingKey() );
+
+		const axisMap = {
+			[ ContainerHelper.DIRECTION_COLUMN ]: 'vertical',
+			[ ContainerHelper.DIRECTION_COLUMN_REVERSED ]: 'vertical',
+			[ ContainerHelper.DIRECTION_ROW ]: 'horizontal',
+			[ ContainerHelper.DIRECTION_ROW_REVERSED ]: 'horizontal',
+			'': isColumnDefault ? 'vertical' : 'horizontal',
+		};
+
+		return axisMap[ currentDirection ];
+	},
```

Comment:
This is related to flexbox, which is not in this task yet AFAIU.

Reply:
I combined both tasks on this ticket. It is more effective but I can split it anyway. @davseve WDYT?

Reply:
At the moment we are creating a regular container. 
We are trying to narrow each task as much as possible. 
We will handle this in a future task.